### DISCUSSION TITLE
Application api access using curl without providing certs

### DISCUSSION THREAD

**Discussion by speeltronics** [gravitational/teleport#10083] [Q&A] [answered]
Is there a way to access an api without providing the certs?

So instead of using this:

curl \
    --cacert /Users/alice/.tsh/keys/teleport.example.com/certs.pem \
    --cert /Users/alice/.tsh/keys/teleport.example.com/alice-app/cluster-name/grafana-x509.pem \
    --key /Users/alice/.tsh/keys/teleport.example.com/alice \
    https://grafana.teleport.example.com:3080/api

Just use this:

curl https://grafana.teleport.example.com:3080/api

If I access my api using a browser, it works fine without providing the certs? If I use curl or postman then it only works when I provide the certs? Why is that?

**Comment by speeltronics** [comment]
For instance, on my browser if I go to my api: https://grafana.teleport.example.com:3080/api
I get the following:
{"errors":{"message":"Unauthorized","stack":{"status":401}}}

On curl though if I try: curl https://grafana.teleport.example.com:3080/api
I get the following:
<a href="https://teleport.example.com:3080/web/launch/grafana.teleport.example.com">Found</a>.

Is there a way to expose the api so I can make api calls without specifying certs? Like I can do with ngrok or inlets? Or is that not the purpose of the appliction api access?

**Comment by webvictim** [comment, marked as answer]
When your browser makes the call to Teleport it's sending cookies along with the request, which lets Teleport know you're authenticated and are allowed to access the application. `curl` doesn't do that, so has to use client certificates instead.

You could try capturing the cookies from your browser and sending them along with the request, but it's not a supported use case.
